source/common/runtime/runtime_impl.cc

```diff
@@ -23,6 +23,29 @@ namespace Runtime {
 
 const size_t RandomGeneratorImpl::UUID_LENGTH = 36;
 
+uint64_t RandomGeneratorImpl::random() {
+  static thread_local uint8_t buffered[2048];
+  static thread_local size_t buffered_idx = sizeof(buffered);
+
+  // Prefetch 2048 bytes of randomness. buffered_idx is initialized to sizeof(buffered),
+  // i.e. out-of-range value, so the buffer will be filled with randomness on the first
+  // call to this function.
+  if (buffered_idx >= sizeof(buffered)) {
+    int rc = RAND_bytes(buffered, sizeof(buffered));
+    ASSERT(rc == 1);
+    UNREFERENCED_PARAMETER(rc);
+    buffered_idx = 0;
+  }
+
+  // Consume sizeof(uint64) bytes from the buffer.
+  ASSERT(buffered_idx + sizeof(uint64_t) <= sizeof(buffered));
+  uint64_t rand;
+  std::memcpy(reinterpret_cast<void*>(&rand), &buffered[buffered_idx], sizeof(uint64_t));
+  buffered_idx += sizeof(uint64_t);
+
+  return rand;
+}
```

Comment:
Can we update `uuid` to use this function (i.e. factor out the `RAND_bytes` pool)?

Reply:
+1, while we are here, it's possible there is a slight perf bump if you used a struct for both thread local variables. I'm not sure of the compiler optimizes this or not or goes through the thread local register multiple times. Not a big deal.

Reply:
This is moot now, because of the migration to array of `uint64_t`.

Reply:
Isn't 16 bytes just 2 `uint64_t` via `random()`?

Reply:
Yes, it is, but refactoring `uuid()` to use `random()` that returns `uint64_t` and stitching source of randomness from 2x `uint64_t` feels more like a hack than a clean interface.

Originally, I considered extracting common code and shared pool of prefetched randomness to `random_bytes()` that returns array of random bytes (see [1d76e26c](https://github.com/envoyproxy/envoy/pull/1465/commits/1d76e26c9857764027436b38eab72cca12a2cbc4)), but that's moot with a change to an array of `uint64_t` in `random()`.

There is also a ~5% performance hit in both cases, due to the extra copies.

FWIW, I'm fine with small performance loss in exchange for a cleaner interface, but the need to stitch source of randomness from 2x `uint64_t` in addition to the small performance loss feels like a bad trade-off.

Reply:
I'm fairly surprised by the performance loss, since the compiler should be able to turn these into register level transfers, but sure, let's just go with what is there now then.